File: backend/unicorn2/src/main/java/com/vortexdbg/arm/backend/Unicorn2Factory.java
package com.vortexdbg.arm.backend;

import com.vortexdbg.Emulator;

import java.io.IOException;

public class Unicorn2Factory extends BackendFactory {

    static {
        try {
            org.scijava.nativelib.NativeLoader.loadLibrary("unicorn");
        } catch (IOException ignored) {
        }
    }

    public Unicorn2Factory(boolean fallbackUnicorn) {
        super(fallbackUnicorn);
    }

    @Override
    protected Backend newBackendInternal(Emulator<?> emulator, boolean is64Bit) {
        return new Unicorn2Backend(emulator, is64Bit);
    }

}

File: backend/unicorn2/src/main/kotlin/com/vortexdbg/arm/backend/Unicorn2Factory.kt
package com.vortexdbg.arm.backend

import com.vortexdbg.Emulator
import java.io.IOException

class Unicorn2Factory(fallbackUnicorn: Boolean) : BackendFactory(fallbackUnicorn) {

    override fun newBackendInternal(emulator: Emulator<*>, is64Bit: Boolean): Backend {
        return Unicorn2Backend(emulator, is64Bit)
    }

    companion object {
        init {
            try {
                org.scijava.nativelib.NativeLoader.loadLibrary("unicorn")
            } catch (ignored: IOException) {
            }
        }
    }

}
